common/src/main/java/com/linecorp/centraldogma/common/PathPatternOptions.java

```diff
@@ -0,0 +1,95 @@
+package com.linecorp.centraldogma.common;
+
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import java.util.function.Function;
+import java.util.regex.Pattern;
+
+import com.google.common.collect.ImmutableSet;
+
+import com.linecorp.centraldogma.internal.Util;
+
+/**
+ * A set of {@link PathPatternOption}s.
+ */
+final class PathPatternOptions {
+
+    private static final Pattern FILE_EXTENSION_PATTERN = Pattern.compile("^.{0,1}[0-9a-z]+$");
+
+    private PathPatternOptions() {}
+
+    /**
+     * Appends "&#47;**" to {@code dirPath}.
+     * Returns the path pattern for matching all file(s) under {@code dirPath}.
+     */
+    public static final Function<String, PathPatternOption> STARTS_WITH = pattern ->
+            new PathPatternOption(pattern,
+                                  dirPath -> {
+                                      checkArgument(Util.isValidDirPath(dirPath), "dir");
+                                      return new DefaultPathPattern(ImmutableSet.of(
+                                              dirPath + (dirPath.endsWith("/") ? "" : "/") + "**"));
```

Comment:
`DefaultPathPattern` does not hold patterns as `Set` so it looks inefficient to create a `Set`. What do you think of creating a constructor for a single string?